classes/order/OrderHistory.php

```diff
@@ -486,11 +486,14 @@ public function sendEmail($order, $template_vars = false)
             ShopUrl::cacheMainDomainForShop($order->id_shop);
 
             $topic = $result['osname'];
+            $carrier = new Carrier((int) $order->id_carrier, $order->id_lang);
```

Comment:
Should maybe be better to check if $carrier `isLoadedObject` before using `$carrier->url` I don't know if an order is always associated to a carrier so adding a check before adding data in `$data` could be more secure.